keras_contrib/applications/resnet.py

```diff
@@ -113,27 +147,45 @@ def _shortcut(input_feature, residual, conv_name_base=None, bn_name_base=None):
     shortcut = input_feature
     # 1 X 1 conv if shape is different. Else identity.
     if stride_width > 1 or stride_height > 1 or not equal_channels:
-        print('reshaping via a convolution...')
+        if verbose:
+            print("    SHORTCUT not right: input -", shortcut.shape, "residual -", residual.shape)
+            print("    reshaping via a convolution...")
         if conv_name_base is not None:
             conv_name_base = conv_name_base + '1'
-        shortcut = Conv2D(filters=residual_shape[CHANNEL_AXIS],
-                          kernel_size=(1, 1),
-                          strides=(stride_width, stride_height),
-                          padding="valid",
-                          kernel_initializer="he_normal",
-                          kernel_regularizer=l2(0.0001),
-                          name=conv_name_base)(input_feature)
+        if time_distributed:
+            shortcut = TimeDistributed(Conv2D(filters=residual_shape[CHANNEL_AXIS],
+                                              kernel_size=(1, 1),
```

Comment:
can there just be one call to Conv2D here and then an if statement that adds TimeDistributed below if enabled?

Reply:
Great suggestion, will implement and squash commits.